javatests/arcs/core/analysis/RecipeGraphFixpointIteratorTest.kt

```diff
@@ -0,0 +1,267 @@
+package arcs.core.analysis
+
+import arcs.core.data.HandleConnectionSpec
+import arcs.core.data.HandleMode
+import arcs.core.data.ParticleSpec
+import arcs.core.data.Recipe
+import arcs.core.data.TypeVariable
+import com.google.common.truth.Truth.assertThat
+import com.google.common.truth.Truth.assertWithMessage
+import org.junit.Assert.assertTrue
+import org.junit.Assert.assertNotNull
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.junit.runners.JUnit4
+
+/** An abstract domain of set, where the elements are ordered by inclusion. */
+class AbstractSet<S>(
+    val value: BoundedAbstractElement<Set<S>>
+) : AbstractValue<AbstractSet<S>> {
+    override val isBottom = value.isBottom
+    override val isTop = value.isTop
+
+    val set: Set<S>? get() = value.value
+
+    constructor(s: Set<S>): this(BoundedAbstractElement.makeValue(s))
+
+    override infix fun isEquivalentTo(other: AbstractSet<S>) =
+        value.isEquivalentTo(other.value) { a, b -> a == b }
+
+    override infix fun join(other: AbstractSet<S>) = AbstractSet(
+        value.join(other.value) { a, b -> a union b }
+    )
+
+    override infix fun meet(other: AbstractSet<S>) = AbstractSet(
+        value.meet(other.value) { a, b -> a intersect b }
+    )
+
+    override fun toString() = when {
+        value.isTop -> "TOP"
+        value.isBottom -> "BOTTOM"
+        else -> "${set}"
+    }
+}
+
+/** Returns the name of the underlying handle or particle. */
+fun RecipeGraph.Node.getName() = when (this) {
+    is RecipeGraph.Node.Particle -> "p:${particle.spec.name}"
+    is RecipeGraph.Node.Handle -> "h:${handle.name}"
+}
+
+/** A simple Fixpoint iterator for the purposes of testing. */
+class TestAnalyzer(
+    val startNames: Set<String>
```

Comment:
I presume in the future this will be figured out from the structure of the graph?

Reply:
Yep, this is just for the tests. The actual analyzer will figure it out from the graph.